File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>


int main(void)
{
    
    string text = get_string("Text :");
    int letters = 0;
    int words = 0;
    int sentences = 0;
    float index;
    float L;
    float S;
    
    int len = strlen(text);
    
    
    //for each character, a loop count if is a letter, a entire word or a sentence
    
    for (int i = 0; i < len; i++)
    {
        //if there is a space, a word is complete
        if (text[i] == ' ')
        {
            words++;
        }
        // at the end of a sentence, count one more.
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
        // for every letter, lower or upcase, count one more
        else if ((text[i] >= 'A' && text[i] <= 'Z') || (text[i] >= 'a' && text[i] <= 'z'))
        {
            letters++;
        }
    }
    words++;
    
    
    // find L and S
    L = (float) letters / words * 100;
    S = (float) sentences / words * 100;
    
    
    // find index
    
    index = 0.0588 * L - 0.296 * S - 15.8;
    
    
    // print the results
    
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else 
    {
        printf("Grade %.0f\n", index);
    }
    
}

File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

# number of sentences on text
sentences = 0
for word in text.split():
    if word[-1] == '.' or word[-1] == '!' or word[-1] == '?':
        sentences += 1

# l: number of letters on text.
l = 0
for letter in text:
    if letter.isalpha():
        l += 1

# L is the average number of letters per 100 words in the text
L = l / len(text.split()) * 100

# S is the average number of sentences per 100 words in the text
S = sentences / len(text.split()) * 100

index = 0.0588 * L - 0.296 * S - 15.8

# prints the result
if index > 16:
    print(f"Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {round(index)}")